From 54acfb45d16716e0d9c28d6bf9e26be5bf71685e Mon Sep 17 00:00:00 2001
From: "Bernhard M. Wiedemann" <bwiedemann@suse.de>
Date: Wed, 9 Jul 2025 12:13:32 +0200
Subject: [PATCH] Avoid randomness in %_tmp_licenses_dir macro

because through
https://github.com/rpm-software-management/rpm/issues/2343
this gets embedded into .src.rpm headers
and disturbs reproducible builds.

This patch was done while working on reproducible builds for openSUSE.
---
 macros.vendored_licenses_packager | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)


diff --git a/macros.vendored_licenses_packager b/macros.vendored_licenses_packager
index 6857c54..9ae6a41 100644
--- a/macros.vendored_licenses_packager
+++ b/macros.vendored_licenses_packager
@@ -15,7 +15,7 @@
 %_set_global_vendor_target_dir() \
 %global __default_vendor_target_dir %{_licensedir}/%{name}/vendored
 
-%_tmp_licenses_dir() "%(mktemp --directory /tmp/tmp_vendored_dir_XXXXXXXXXX)"
+%_tmp_licenses_dir() "%_builddir/tmp_vendored_dir"
 
 %_set_tmp_licenses_vendored_dir() \
 %global __tmp_licenses_vendored_dir %{_tmp_licenses_dir}
